pages/developers.tsx

```diff
@@ -37,49 +49,43 @@ const options_time = [
 export default function PageDevelopers() {
   const {dispatch, state: {loading}} = useContext(ApplicationContext);
   const [issues, setIssues] = useState<IssueData[]>([]);
-  const [filterStateIssues, setfilterStateIssues] = useState({
-    state: "",
-    issues: [],
-  });
-
+  const [filterByState, setFilterByState] = useState<Filter>(filtersByIssueState[0]);
 
-  function handleChangeFilterIssue(params: { value: string; label: string; }) {
-    if (params.value === "all") {
-      setfilterStateIssues({
-        state: params.value,
-        issues,
-      });
-    } else {
-      setfilterStateIssues({
-        state: params.value,
-        issues: issues.filter(
-          (obj: IssueData) => obj.state.toLowerCase() === params.value
-        ),
-      });
-    }
+  function handleChangeFilterByState(filter: Filter) {
+    setFilterByState(filter);
   }
 
   function updateIssuesList(issues: IssueData[]) {
     console.log(`got issues`, issues);
     setIssues(issues);
-    if (filterStateIssues.issues.length === 0) {
-      setfilterStateIssues({ state: "all", issues });
-    }
   }
 
-  function getIssues() {
-    dispatch(changeLoadState(true))
-    GithubMicroService.getIssues()
-                      .then(updateIssuesList)
-                      .catch((error) => {
-                        console.log('Error', error)
-                      })
-                      .finally(() => {
-                        dispatch(changeLoadState(false))
-                      });
-  }
+  useEffect(() => {
+    async function getIssues() {
+      dispatch(changeLoadState(true))
+      GithubMicroService.getIssues()
+                        .then(updateIssuesList)
+                        .catch((error) => {
+                          console.log('Error', error)
+                        })
+                        .finally(() => {
+                          dispatch(changeLoadState(false))
+                        });
+    }
+    getIssues();
+  }, []);
+
+  const isDraftIssue = (issue: IssueData) => issue.state === 'draft';
+  const isClosedIssue = (issue: IssueData) => issue.state === 'closed';
+  const isOpenIssue = (issue: IssueData) => !isDraftIssue(issue) && !isClosedIssue(issue);
 
-  useEffect(getIssues, []);
+  const issuesFilteredByState = issues.filter(issue => {
```

Comment:
Please, use a function statement instead of a fat arrow constant to define functions. 

PS: _this_ is the place you could have written those constants as closures inside this function as it's not used anywhere else not would it hamper the readability of the code -- in fact: it would signal to the next developer that this 3 functions are only used inside that function.